#ifndef USERSMANAGERDIALOG_HPP
#define USERSMANAGERDIALOG_HPP

#include <QDialog>
#include <QMessageBox>
#include <QSqlDatabase>

#include "ui_UsersManager.h"
#include "UserContainer.hpp"
#include "Persistence/UserDbInterface.hpp"

class UsersManagerDialog : public QDialog
{
    Q_OBJECT
private:
    Ui::UsersManager *ui;
    UserContainer *userContainer;
    UserDbInterface *userDb;

    bool updated;
public:
    UsersManagerDialog(UserContainer *userContainer, UserDbInterface*userDb, QWidget *parent=0);
    ~UsersManagerDialog();
    void addUserToList(const QString &name);
public slots:
    void open();

private slots:
    void addUser();
    void deleteUser();
    void dialogClosed();
signals:
    //Esta se침al es emitida cuando se cierra el di치logo y los usuarios han
    //cambiado: se ha a침adido o eliminado alguno por lo que ser치 necesario
    //notificarlo.
    void closedAndUsersChaned();
};

#endif // USERSMANAGERDIALOG_HPP
